google/data_source_google_billing_account.go

```diff
@@ -0,0 +1,111 @@
+package google
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"google.golang.org/api/cloudbilling/v1"
+	"google.golang.org/api/googleapi"
+)
+
+func dataSourceGoogleBillingAccount() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceBillingAccountRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"display_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"open": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"project_ids": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func dataSourceBillingAccountRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	name, nameOk := d.GetOk("name")
+	displayName, displayNameOk := d.GetOk("display_name")
+	if nameOk == displayNameOk {
+		return fmt.Errorf("One of ['name', 'display_name'] must be set to read billing accounts")
+	}
+
+	var billingAccount *cloudbilling.BillingAccount
+	if nameOk {
+		resp, err := config.clientBilling.BillingAccounts.Get(name.(string)).Do()
+		if err != nil {
+			if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == http.StatusNotFound {
+				return fmt.Errorf("Billing account not found: %s", name)
+			}
+
+			return fmt.Errorf("Error reading billing account: %s", err)
+		}
+
+		billingAccount = resp
+	} else {
+		resp, err := config.clientBilling.BillingAccounts.List().Do()
```

Comment:
This call has pagination.

Here is an example of how we page through results in other parts of the code:
https://github.com/terraform-providers/terraform-provider-google/blob/master/google/resource_google_project_services.go#L171

Reply:
Done.